apps/dashboard/app/models/project.rb

```diff
@@ -1,59 +1,110 @@
 # frozen_string_literal: true
 
-
 class Project
   include ActiveModel::Model
+  include ActiveModel::Validations
 
   class << self
     def all
-      # return [Array] of all projects in ~/ondemand/data/sys/projects
       return [] unless dataroot.directory? && dataroot.executable? && dataroot.readable?
 
       dataroot.children.map do |d|
-        Project.new({ :dir => d.basename })
+        Project.new({ :name => d.basename })
       rescue StandardError => e
         Rails.logger.warn("Didn't create project. #{e.message}")
         nil
       end.compact
     end
 
     def find(project_path)
+      Rails.logger.debug("project_path: #{project_path}")
       full_path = dataroot.join(project_path)
+      Rails.logger.debug("full_path: #{full_path}")
       return nil unless full_path.directory?
 
-      Project.new({ dir: full_path })
+      Project.new({ name: full_path.basename })
     end
 
     def dataroot
-      Rails.logger.debug("project path is: #{OodAppkit.dataroot.join('projects')}")
-
-      OodAppkit.dataroot.join('projects').tap do |p|
-        p.mkpath unless p.exist?
+      OodAppkit.dataroot.join('projects').tap do |path|
+        path.mkpath unless path.exist?
       rescue StandardError => e
         Pathname.new('')
       end
     end
   end
 
-  attr_reader :dir
+  validates :directory, presence: true
+  validates :directory, format: {
+    with: /\A[\w-]+\z/,
+    message: 'Directory may only contain letters, digits, dashes, and underscores'
+  }
+
+  attr_reader :directory, :description
+  
+  delegate :icon, :name, :description, to: :manifest
 
-  def initialize(args = {})
-    @dir = args.fetch(:dir, nil).to_s
+  def initialize(attributes = {})
+    @name    = attributes.fetch(:name, nil).to_s
+    @description  = attributes.fetch(:description, nil).to_s
+  end
+ 
+  # @params [Hash] 
+  # @return [Bool]
+  def save(attributes)
+    result = update(attributes)
+    result
   end
 
-  def config_dir
-    Pathname.new("#{absolute_dir}/.ondemand").tap { |p| p.mkpath unless p.exist? }
+  # @params [Hash] 
+  # @return [Bool]
+  def update(attributes)
+    # only have side effects in update
+    new_manifest = Manifest.load(manifest_path)
+    new_manifest = manifest.merge(attributes)
+    # all manifest.valid? does is check if .to_h works on object passed
+    # need to check that the name accepts only letters, underscore, dash, digits
+    if attributes_valid?(attributes)
+      new_manifest.valid? ? new_manifest.save(manifest_path) : false
+    else
+      Rails.logger.error("Attributes passed to manifest invalid")
+      false
+    end
```

Comment:
I need to pass the failed validation back here. Getting empty hash at the moment.